src/firefly/js/charts/ChartUtil.js

```diff
@@ -288,7 +288,7 @@ export function getOptionsUI(chartId) {
     const {data, fireflyData, activeTrace=0} = getChartData(chartId);
     const type = get(data, [activeTrace, 'type'], 'scatter');
     const dataType = get(fireflyData, [activeTrace, 'dataType'], '');
-    if (isScatter2d(type)) {
+    if (dataType === 'fireflyScatter' || isScatter2d(type)) {
```

Comment:
in here, either 'fireflyScatter' or scatter2d returns ScatterOptions. 
Question, for function getToolBarUI() next to this function, isn't 'fireflyScatter' treated the same as 'isScatter2d'? 
Also there is no 'selection box'  shown up for 'fireflyScatter' chart.

Reply:
Thank you for finding this issue. Indeed, I made a change that for firefly enhanced charts plotly trace type is not assigned until the data are retrieved. This bug should be fixed in the latest commit.